### ISSUE TITLE
feature: Be able to define rules to run webhook

### ISSUE THREAD

**Issue by BeArchiTek** [opsmill/infrahub#4012]
### Component

Frontend UI, API Server / GraphQL

### Describe the Feature Request

As a user, I want to be able to choose for which event and objects I want to run a webhooks (custom or not) 

### Describe the Use Case

Right now, If I create a device with 24 interfaces, I will trigger the webhook 25 times.

I want to run it only once when a model of my choosing is created.

Link to https://github.com/opsmill/infrahub/issues/3286

### Additional Information

Example in Netbox : 
https://demo.netbox.dev/extras/event-rules/
![image](https://github.com/user-attachments/assets/f87f0a88-c959-494a-95e1-70c024002e9f)

**Comment by ogenstad** [comment]
Related to #1770 (depending on how we implement this)

**Comment by dgarros** [comment]
I believe most of it has been implemented in 1.2
Please create a new issue if something is missing
